### PR TITLE
[NEW] Custom login wallpapers

### PR DESCRIPTION
**Pull request by vynmera**
This PR allows people to customize their Rocket.Chats a bit more with a custom login wallpaper.

**Right now it looks like this...**

**But by tweaking this new option in Assets...**

**...it can look like this!**

### CODE REVIEW

**Comment by rauhmaru** [comment]
AWESOME!!!

**Comment by ggazzo** [comment]
@vynmera great feature! but I think you forgot some screens 

image

File: packages/rocketchat-ui-login/client/login/layout.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <template name="loginLayout">
-	<section class="rc-old full-page color-tertiary-font-color">
+	<section class="rc-old full-page color-tertiary-font-color" style="background-image: url('{{ backgroundUrl }}')">
```
Comment (ggazzo) [inline packages/rocketchat-ui-login/client/login/layout.html]: what do you think about 
```html
```

File: packages/rocketchat-ui-login/client/login/layout.js [hunk 2]
```diff
@@ -1,3 +1,13 @@
 Template.loginLayout.onRendered(function() {
 	$('#initial-page-loading').remove();
 });
+
+Template.loginLayout.helpers({
+	backgroundUrl() {
+		const asset = RocketChat.settings.get('Assets_background');
+		const prefix = __meteor_runtime_config__.ROOT_URL_PATH_PREFIX || '';
+		if (asset != null) {
```
[the remark this answers is not in the document]
Comment (vynmera) [reply packages/rocketchat-ui-login/client/login/layout.js]: Prefix might not be set on all servers, so I'll leave that out

**Comment by vynmera** [comment]
This should fix the requested changes as much as possible:

@engelgabriel Register Username is now styled.
@ggazzo I had to change the syntax a bit and remove the `prefix` from your suggested `if`. Please rereview and tell me what you think :-)

**Comment by vynmera** [comment]
@kaiiiiiiiii Spacebars _really_ doesn't like it when you do that, apparently. Doing it like this does seem to remove the style tag from DOM when unused though (unless Firefox is lying to me)

File: packages/rocketchat-ui-login/client/username/layout.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+Template.usernameLayout.helpers({
+	backgroundUrl() {
+		const asset = RocketChat.settings.get('Assets_background');
+		const prefix = __meteor_runtime_config__.ROOT_URL_PATH_PREFIX || '';
+		if (asset && (asset.url || asset.defaultUrl)) {
+			return `${ prefix }/${ asset.url || asset.defaultUrl }`;
+		}
+	}
+});
```
[the remark this answers is not in the document]
Comment (ggazzo) [reply packages/rocketchat-ui-login/client/username/layout.js]: 😄

[the remark this answers is not in the document]
Comment (vynmera) [reply packages/rocketchat-ui-login/client/username/layout.js]: what did it mean by this? 🤔

[the remark this answers is not in the document]
Comment (vynmera) [reply packages/rocketchat-ui-login/client/username/layout.js]: img_20180612_114539
What's it talking about? I'm not at home right now but GH clearly shows a 10th line..?

[the remark this answers is not in the document]
Comment (ggazzo) [reply packages/rocketchat-ui-login/client/username/layout.js]: good question :x I dont know if the github insert the last line... could you check on your editor?

[the remark this answers is not in the document]
Comment (vynmera) [reply packages/rocketchat-ui-login/client/username/layout.js]: @ggazzo 

Eslint is happy for me too...?

[the remark this answers is not in the document]
Comment (vynmera) [reply packages/rocketchat-ui-login/client/username/layout.js]: no u

[the remark this answers is not in the document]
Comment (kaiiiiiiiii) [reply packages/rocketchat-ui-login/client/username/layout.js]: The codacy-bot is really a pain in the arse, isn’t it? :D

[the remark this answers is not in the document]
Comment (vynmera) [reply packages/rocketchat-ui-login/client/username/layout.js]: there is clearly a newline >:(

[the remark this answers is not in the document]
Comment (vynmera) [reply packages/rocketchat-ui-login/client/username/layout.js]: are you high

**Comment by localguru** [comment]
@vynmera could you share the background image. :)

**Pull request merged**